src/ripple/conditions/hashlock.h

```diff
@@ -0,0 +1,111 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_CONDITIONS_HASHLOCK_H
+#define RIPPLE_CONDITIONS_HASHLOCK_H
+
+#include <ripple/basics/base_uint.h>
+#include <ripple/conditions/condition.h>
+#include <ripple/conditions/fulfillment.h>
+#include <ripple/conditions/impl/base64.h>
+#include <ripple/protocol/digest.h>
+#include <boost/regex.hpp>
+#include <boost/optional.hpp>
+#include <algorithm>
+#include <iterator>
+#include <vector>
+
+namespace ripple {
+namespace cryptoconditions {
+
+struct hashlock_t final
```

Comment:
We tend to use `final` very sparingly. Unless this _needs_ to be `final` it probably fits in better with rippled if we remove the keyword.